Actions/AL-Go-Helper.ps1

```diff
@@ -645,6 +645,75 @@ function ExcludeUnneededApps {
     }
 }
 
+function ResolveProjectFolders {
+    Param(
+        [string] $project,
+        [string] $baseFolder,
+        [ref] $projectSettings
+    )
+    $projectPath = Join-Path $baseFolder $project -Resolve
+
+    Push-Location $projectPath
+
+    try {
+        # Filter out app folders that doesn't contain an app.json file
+        'appFolders', 'testFolders', 'bcptTestFolders' | ForEach-Object {
+            $folders = $projectSettings.Value.$_
+            
+            $resolvedPaths = @()
+            foreach($folder in $folders) {
+                $aLProjectFolder = Join-Path $projectPath $folder
+                $resolvedPaths += Resolve-Path $aLProjectFolder -ErrorAction Ignore | Where-Object { Test-Path (Join-Path $_ 'app.json') }
+            }
+
+            # Get the name of the folder for each resolved path
+            $projectSettings.Value.$_ = @($resolvedPaths | ForEach-Object { $(Get-Item $_).Name })
+        }
+
+
+        # If no app folders are specified in AL-Go settings, check for AL apps in the project folder
+        if (!$projectSettings.Value.appFolders -and !$projectSettings.Value.testFolders -and !$projectSettings.Value.bcptTestFolders) {
+            Get-ChildItem -Path $projectPath -Recurse | Where-Object { $_.PSIsContainer -and (Test-Path -Path (Join-Path $_.FullName "app.json")) } | ForEach-Object {
+                $aLProjectFolder = $_
+                $appJson = Get-Content (Join-Path $aLProjectFolder.FullName "app.json") -Encoding UTF8 | ConvertFrom-Json
+
+                $isTestApp = $false
+                $isBcptTestApp = $false
+                
+                # if an AL app has a dependency to a test app, it is a test app
+                # if an AL app has a dependency to an app from the performance toolkit apps, it is a bcpt test app
+                if ($appJson.PSObject.Properties.Name -eq "dependencies") {
+                    $appJson.dependencies | ForEach-Object {
+                        $id = $_.Id
+                        if ($_.PSObject.Properties.Name -eq "AppId") {
+                            $id = $_.AppId
+                        }
+
+                        # Check if the app is a test app or a bcpt app
+                        if ($performanceToolkitApps.Contains($id)) { 
+                            $isBcptTestApp = $true
+                        }
+                        elseif ($testRunnerApps.Contains($id)) { 
+                            $isTestApp = $true
+                        }
+                    }
+                    
+                }
+
+                switch ($true) {
+                    $isTestApp { $projectSettings.Value.testFolders += @($aLProjectFolder.Name) }
+                    $isBcptTestApp { $projectSettings.Value.bcptTestFolders += @($aLProjectFolder.Name) }
+                    Default { $projectSettings.Value.appFolders += @($aLProjectFolder.Name) }
+                }
+            }
+        }
+    }
+    finally {
```

Comment:
Consider adding a catch and writing the error to the terminal? If something goes wrong it might be easier to debug that way

Reply:
If an error is thrown, it is not swallowed, so it'll bubble up in the log.